src/middlewared/middlewared/plugins/jail.py

```diff
@@ -989,6 +1006,103 @@ def rc_action(self, action):
 
         return True
 
+    @accepts(Dict('props', additional_attrs=True))
+    def update_defaults(self, props):
+        """
+        Update default properties for iocage which will remain true for all jails moving on i.e nat_backend
+        """
+        iocage = ioc.IOCage(jail='default', reset_cache=True)
+        for prop in props:
+            iocage.set(f'{prop}={props[prop]}')
+
+    @private
+    def retrieve_default_iface(self):
+        default_ifaces = ioc_common.get_host_gateways()
+        if default_ifaces['ipv4']['interface']:
+            return default_ifaces['ipv4']['interface']
+        elif default_ifaces['ipv6']['interface']:
+            return default_ifaces['ipv6']['interface']
+
+    @private
+    def retrieve_vnet_interface(self, iface):
+        if iface == 'auto':
+            return self.retrieve_default_iface()
+        elif iface != 'none':
+            return iface
+        else:
+            return None
+
+    @private
+    def retrieve_vnet_bridge(self, interfaces):
+        bridges = []
+        for interface in interfaces.split(','):
+            if interface.split(':')[-1] not in bridges:
+                bridges.append(interface.split(':')[-1])
+        return bridges
+
+    @private
+    def retrieve_nat_interface(self, iface):
+        return self.retrieve_default_iface() if iface == 'none' else iface
+
+    @private
+    async def nic_capability_checks(self, jails=None):
+        """
+        For vnet/nat based jails, when jail is started, if NIC has certain capabilities set, we experience a hiccup
+        in the network traffic which for failover can result in backup node coming online. This method returns
+        interfaces which will be affected by this based on the jails user has.
+        """
+        jail_nics = defaultdict(set)
+        system_ifaces = {i['name']: i for i in await self.middleware.call('interface.query')}
+        for jail in await self.middleware.call(
+            'jail.query', [['OR', [['vnet', '=', 1], ['nat', '=', 1]]]]
+        ) if not jails else jails:
+            nic = await self.middleware.call(
+                f'jail.retrieve_{"nat" if jail["nat"] else "vnet"}_interface',
+                jail['nat_interface' if jail['nat'] else 'vnet_default_interface']
+            )
+            if nic in system_ifaces:
+                conflicts = {'TSO4', 'TSO6', 'VLAN_HWTSO'} if jail['nat'] else {
```

Comment:
Do we lose some features / performance when disabling this? Can these capabilities be re-enabled later after jail / VM network configuration?

Reply:
Yes there is a performance regression. If i got you correctly @william-gr, we don't want to re-enable those when these jails/vms have been deleted ?

Reply:
I think If Jails/VMs are deleted we should somehow enable these flags again

Reply:
@william-gr @sonicaj now when I understand what's going on maybe we should add disabling these capabilities to "interface options" and just forbid creating jails / vms using interfaces with capabilities not disabled?

Reply:
Users will have a better understanding on what's going on, why and when, and we won't need to remember to re-enable capabilities again. Explicit is better than implicit.

Reply:
@william-gr @themylogin's suggestion makes sense to me. Do you think we should correct this with that approach ? We can store capabilities in the database so that we can configure them on system boot before we start jails/vms so they are correctly started

Reply:
Yes, I like that idea, lets do that. But I think we should also add a way to do that "automatically", warning the user what we are about to do and how, when creating a jail/vm.

Its hard to tell the user to go to another screen and do X, Y and Z before he can actually perform what he wanted in the first place.

Reply:
Can you update the docstring here?

Instead of `we experience a hiccup in the network traffic which for failover can result in backup node coming online.` Just make it `we experience a hiccup in the network traffic which can cause a failover to occur.`